src/atomate2/vasp/flows/md.py

```diff
@@ -0,0 +1,166 @@
+
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING
+
+from jobflow import Flow, Maker, OutputReference
+
+from atomate2.vasp.jobs.md import MDMaker, md_output
+from atomate2.vasp.sets.core import MDSetGenerator
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from pymatgen.core import Structure
+
+    from atomate2.vasp.jobs.base import BaseVaspMaker
+
+
+@dataclass
+class MultiMDMaker(Maker):
+    """
+    Maker to perform an MD run split in several steps.
+
+    Parameters
+    ----------
+    name : str
+        Name of the flows produced by this maker.
+    md_makers : .BaseVaspMaker
+        Maker to use to generate the first relaxation.
+    """
+
+    name: str = "multi md"
+    md_makers: list[BaseVaspMaker] = field(default_factory=lambda: list(MDMaker()))
+
+    def make(
+        self,
+        structure: Structure,
+        prev_dir: str | Path | None = None,
+        prev_traj_ids: list[str] | None = None,
+    ):
+        """
+        Create a flow with several chained MD runs.
+
+        Parameters
+        ----------
+        structure : .Structure
+            A pymatgen structure object.
+        prev_dir : str or Path or None
+            A previous VASP calculation directory to copy output files from.
+        prev_traj_ids: a list of ids of job identifying previous steps of the
+            MD trajectory.
+
+        Returns
+        -------
+        Flow
+            A flow containing n_runs MD calculations.
+        """
+        md_job = None
+        md_jobs = []
+        for i, maker in enumerate(self.md_makers, 1):
+            if md_job is None:
+                md_structure = structure
+                md_prev_dir = prev_dir
+            else:
+                md_structure = md_job.output.structure
+                md_prev_dir = md_job.output.dir_name
+            md_job = maker.make(md_structure, prev_dir=md_prev_dir)
+            md_job.name += f" {i}"
+            md_jobs.append(md_job)
+
+        output_job = md_output(
+            structure=md_jobs[-1].output.structure,
+            vasp_dir=md_jobs[-1].output.dir_name,
+            traj_ids=[j.uuid for j in md_jobs],
+            prev_traj_ids=prev_traj_ids,
+        )
+        output_job.name = "molecular dynamics output"
+
+        md_jobs.append(output_job)
+
+        return Flow(md_jobs, output_job.output, name=self.name)
+
+    def restart_from_uuid(self, md_ref: str | OutputReference):
+        """
+        Create a flow from the output reference of another MultiMDMaker.
+
+        The last output will be used as the starting point and the reference to
+        all the previous steps will be included in the final document.
+
+        Parameters
+        ----------
+        md_ref: str or OutputReference
+            The reference to the output of another MultiMDMaker
+
+        Returns
+        -------
+            A flow containing n_runs MD calculations.
+        """
+        if isinstance(md_ref, str):
```

Comment:
Would be good to add a test for the `isinstance(md_ref, str)` case

Reply:
I added a test specific for this method. I did not add a new full Flow test, as it will be equivalent to the one already tested.